R/Switzerland.R

```diff
@@ -0,0 +1,122 @@
+#' for Liechtenstein as `FL-FL`, consistent with the Swiss ISO 3166-2 codes
+#' which are of the form `CH-BE`, `CH-ZH`, `CH-VD`, ...
+#'
+#' If you do not wish to work with Liechtenstein
+#' data, filter out on this code. Note that this is labelled as a ISO 3166-2
+#' code but Liechtenstein's real ISO 3166-2 codes refer to sub-national
+#' regions.
+#'
+#' @section Additional data:
+#'
+#' In addition to the standard `covidregionaldata` columns provided,
+#' the OpenDataZH source data provides other figures for ICU occupancy,
+#' number of patients on ventilators, and the how many individuals are
+#' isolated or quarantined. These columns are passed through unchanged.
+
+#' Further detail on them can be found at
+# nolint start
+#' \url{https://github.com/openZH/covid_19/#swiss-cantons-and-principality-of-liechtenstein-unified-dataset}
+#' @source \url{https://github.com/openZH/covid_19/}
+# nolint end
+#'
+#' @export
+#' @concept dataset
+#' @family subnational
+#' @examples
+#' \dontrun{
+#' region <- Switzerland$new(verbose = TRUE, steps = TRUE, get = TRUE)
+#' region$return()
+#' }
+Switzerland <- R6::R6Class("Switzerland",
+  inherit = DataClass,
+  public = list(
+
+    # Core Attributes
+    #' @field origin name of origin to fetch data for
+    origin = "Switzerland",
+    #' @field supported_levels A list of supported levels.
+    supported_levels = list("1"),
+    #' @field supported_region_names A list of region names in order of level.
+    supported_region_names = list("1" = "canton"),
+    #' @field supported_region_codes A list of region codes in order of level.
+    supported_region_codes = list("1" = "iso_3166_2"),
+    #' @field common_data_urls List of named links to raw data.
+    # nolint start
+    common_data_urls = list(
+      "main" = "https://github.com/openZH/covid_19/raw/master/COVID19_Fallzahlen_CH_total_v2.csv"
+    ),
+    # nolint end
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c(
+      "hosp_new",
+      "deaths_total",
+      "recovered_total",
+      "cases_total",
+      "tested_total"
+    ),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble
+    set_region_codes = function() {
+      self$codes_lookup$`1` <- tibble(
+        code = c(
+          "CH-AG", "CH-AR", "CH-AI", "CH-BL", "CH-BS",
+          "CH-BE", "CH-FR", "CH-GE", "CH-GL", "CH-GR", "CH-JU", "CH-LU",
+          "CH-NE", "CH-NW", "CH-OW", "CH-SG", "CH-SH", "CH-SZ", "CH-SO",
+          "CH-TG", "CH-TI", "CH-UR", "CH-VS", "CH-VD", "CH-ZG", "CH-ZH",
+          "FL-FL"
+        ),
+        region = c(
+          "Aargau", "Appenzell Ausserrhoden",
+          "Appenzell Innerrhoden", "Basel-Landschaft",
+          "Basel-Stadt", "Bern",
+          "Fribourg", "Gen\u00e8ve",
+          "Glarus", "Grischun",
+          "Jura", "Luzern", "Neuch\u00e2tel", "Nidwalden",
+          "Obwalden", "Sankt Gallen", "Schaffhausen",
```

Comment:
```suggestion
          "Obwalden", "St. Gallen", "Schaffhausen",
```

Sankt is usually abbreviated as "St.", not sure if having a period sign has any drawbacks.